build/repo.targets

```diff
@@ -107,69 +111,33 @@
   </Target>
 
   <Target Name="BuildRepositories"
-     DependsOnTargets="_FilterRepositories;_FindDotNetPath;_GenerateRestoreGraphSpecs;_GenerateBuildGraph;_UpdateNuGetConfig;_CreateRepositoriesListWithCommits">
-
-    <PropertyGroup>
-      <!-- If there are duplicate properties, the properties which are defined later in the order would override the earlier ones -->
-      <RepositoryBuildArguments>$(RepositoryBuildArguments) /p:BuildNumber=$(BuildNumber) /p:Configuration=$(Configuration)</RepositoryBuildArguments>
-
-      <_BuildRepositoryProperties>
-        UniverseBuildDir=$(BuildDir);
-        UniverseMSBuildDir=$(ArtifactsDir)msbuild;
-        BuildInParallel=$(BuildInParallel);
-        Configuration=$(Configuration);
-        DotNetPath=$(DotNetPath);
-        KoreBuildDirectory=$(MSBuildProjectDirectory)\;
-        KoreBuildProject=$(MSBuildProjectFile);
-        RepositoryRoot=$(RepositoryRoot);
-        _BuildGraphFile=$(_BuildGraphFile);
-        _CloneRepositoryRoot=$(_CloneRepositoryRoot);
-        _DependencyPackagesDirectory=$(_DependencyPackagesDirectory);
-        _RepositoryBuildTargets=$(_RepositoryBuildTargets);
-        RepositoryBuildArguments=$(RepositoryBuildArguments);
-        _RestoreGraphSpecsDirectory=$(_RestoreGraphSpecsDirectory);
-        PackagePublisherPath=$(PackagePublisherNetCoreApp)
-      </_BuildRepositoryProperties>
-
-      <_BuildRepositoryProperties Condition="'$(PublishPackages)'=='true'">
-        $(_BuildRepositoryProperties);
-        APIKey=$(APIKey);
-        NuGetPublishVolatileFeed=$(NuGetPublishVolatileFeed);
-        PublishPackages=$(PublishPackages)
-      </_BuildRepositoryProperties>
-    </PropertyGroup>
-
-    <MSBuild
-      Projects="$(MSBuildThisFileDirectory)RepositoryBuild.targets"
-      Targets="BuildRepositories"
-      Properties="$(_BuildRepositoryProperties)" />
-  </Target>
+     DependsOnTargets="_PrepareRepositories;_FindDotNetPath;_CreateRepositoriesListWithCommits;_UpdateNuGetConfig;_GenerateBuildGraph;_BuildRepositories" />
 
-  <Target Name="_GenerateRestoreGraphSpecs" DependsOnTargets="_FindDotNetPath">
+  <Target Name="_PrepareRestoreGraphSpecs" DependsOnTargets="_PrepareRepositories">
     <ItemGroup>
       <Solution Include="$(_CloneRepositoryRoot)%(Repository.Identity)\*.sln">
         <Repository>%(Repository.Identity)</Repository>
       </Solution>
 
+      <RestoreGraphOutputs Include="$(_RestoreGraphSpecsDirectory)%(Solution.Repository)\%(Solution.FileName)%(Solution.Extension).json" />
+
       <Solution>
         <AdditionalProperties>RestoreGraphOutputPath=$(_RestoreGraphSpecsDirectory)%(Solution.Repository)\%(Solution.FileName)%(Solution.Extension).json</AdditionalProperties>
       </Solution>
     </ItemGroup>
+  </Target>
 
+  <Target Name="_GenerateRestoreGraphSpecs" DependsOnTargets="_PrepareRestoreGraphSpecs" Inputs="@(Solution)" Outputs="@(RestoreGraphOutputs)">
```

Comment:
This looks buggy. Solution files can stay the same even when the repository dependency graph changes.